tests/unittest/_torch/auto_deploy/unit/singlegpu/models/test_moe_weight_loading.py

```
@@ -0,0 +1,162 @@
+    """Module with 4 experts and one fused weight (single attr)."""
+    module = _make_moe_module(num_experts=4)
+    setup_fused_moe_weights(module, module.experts, {"weight": "_fused_weight"})
+    return module
+
+
+def test_moe_load_pre_stacked_format(moe_module):
+    """Loading a checkpoint that already has the fused weight works."""
+    stacked = torch.stack([moe_module.experts[i].weight.data.clone() for i in range(4)], dim=0)
+    state_dict = {"_fused_weight": stacked}
+    moe_module.load_state_dict(state_dict, strict=False)
+    fused = getattr(moe_module, "_fused_weight")
```

Ruff (0.15.4)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)